torch/csrc/jit/codegen/cuda/kernel_cache.h

```diff
@@ -56,10 +56,34 @@ class TORCH_CUDA_CU_API FusionKernelRuntime {
     }
   }
 
+  //! query if we already have a compiled kernel for execution
+  bool isCompiled() {
+    std::unique_lock<std::mutex> lock0(mutex_, std::try_to_lock);
+    std::unique_lock<std::mutex> lock1(compiling_, std::try_to_lock);
+    if (!lock0.owns_lock() || !lock0.owns_lock()) {
+      // compilation in progress
+      return false;
+    }
+
+    for (const auto& executor : executors_) {
+      if (!executor.compiled()) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  //! starts compilation async
+  void compileAsync(KernelArgumentHolder& inputs);
+
+  //! maps entries in `args` to fusion inputs.
+  //! Note that this function also pushes extra bits like dimension extent into `args` for expression evaluator binding. So consider your `args` poluted after this function and use it with caution.
```

Comment:
If you look at the bottom of the function, we are pushing dimension extent into args, simply because there's no good place to hold it otherwise. `args.push(tensor_arg_abstract->getSize(dim));`

 So after the mapping function, `args` is not what is was, which should be expected, since it's passes as a non-const reference to the function :wink:

Reply:
So the only point you were making is `mapFusionInputsToArgs` actually modifies `KernelArgumentHolder& args` by filling in the input information.

Reply:
Yep.